File: android/app/src/main/java/com/dshx/game/su/ui/screens/NewspaperScreen.kt
package com.dshx.game.su.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.dshx.game.su.AppState
import com.dshx.game.su.Config
import com.dshx.game.su.GameData
import com.dshx.game.su.NewsItem
import com.dshx.game.su.Sfx
import com.dshx.game.su.ui.theme.LocalGameFont
import com.dshx.game.su.ui.toColor
import com.dshx.game.su.world.World
import kotlin.math.floor

// ============================================================================
// NewspaperScreen — 城市简报，与 scripts/Screens/NewspaperScreen.lua 1:1 对应
// ============================================================================

@Composable
fun NewspaperContent() {
    val C = Config.COLORS
    val s = GameData.current
    val level = World.cityLevel()
    val live = AppState.liveTick

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(C.uiBackdrop.toColor())
    ) {
        // 顶栏
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(start = 14.dp, end = 14.dp, top = 12.dp, bottom = 12.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            Row(
                modifier = Modifier
                    .background(C.panelWhite.toColor(), RoundedCornerShape(18.dp))
                    .clickable {
                        Sfx.play("sfx_click", 0.6f)
                        AppState.screen = "map"
                    }
                    .padding(horizontal = 14.dp, vertical = 8.dp),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(4.dp)
            ) {
                Text(
                    "← 返回地图", fontSize = 13.sp, fontWeight = FontWeight.Bold,
                    color = C.textDark.toColor(), fontFamily = LocalGameFont.current
                )
            }
            Text(
                (s?.cityName ?: "晨光市") + "简报", fontSize = 16.sp, fontWeight = FontWeight.Bold,
                color = C.textDark.toColor(), fontFamily = LocalGameFont.current
            )
            Box(modifier = Modifier.width(84.dp))   // 占位平衡
        }

        // 概览卡
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(start = 14.dp, end = 14.dp, bottom = 8.dp)
                .background(C.panelWhite.toColor(), RoundedCornerShape(16.dp))
                .padding(horizontal = 16.dp, vertical = 12.dp)
        ) {
            if (s != null) {
                Text(
                    level.name + " · " + GameData.monthLabel() + " 概况",
                    fontSize = 13.sp, fontWeight = FontWeight.Bold,
                    color = C.textDark.toColor(), fontFamily = LocalGameFont.current,
                    modifier = Modifier.padding(bottom = 6.dp)
                )
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween
                ) {
                    Text(
                        "财政 ¥" + floor(s.funds).toInt() + "万", fontSize = 12.sp,
                        color = C.accentGold.toColor(), fontFamily = LocalGameFont.current
                    )
                    Text(
                        "人口 " + s.population.toInt(), fontSize = 12.sp,
                        color = C.textDark.toColor(), fontFamily = LocalGameFont.current
                    )
                    Text(
                        "满意度 " + floor(s.happiness).toInt(), fontSize = 12.sp,
                        color = if (s.happiness >= 60) C.accentGreen.toColor() else C.accentRed.toColor(),
                        fontFamily = LocalGameFont.current
                    )
                }
            }
        }

        // 新闻滚动
        Column(
            modifier = Modifier
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
                .padding(start = 14.dp, end = 14.dp, top = 4.dp, bottom = 16.dp),
            verticalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            val news = GameData.recentNews(20)
            if (news.isEmpty()) {
                NewsCard(
                    NewsItem(GameData.monthLabel(), "暂无报道", "城市还在建设中。", "头条")
                )
            } else {
                for (i in news.indices.reversed()) {
                    NewsCard(news[i])
                }
            }
        }
    }
}

@Composable
private fun NewsCard(n: NewsItem) {
    val C = Config.COLORS
    val urgent = n.tag == "头条"
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(C.chipBg.toColor(), RoundedCornerShape(12.dp))
            .border(1.dp, if (urgent) C.accentRed.toColor() else C.border2.toColor(), RoundedCornerShape(12.dp))
            .padding(horizontal = 12.dp, vertical = 10.dp)
    ) {
        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
            Text(
                n.tag, fontSize = 10.sp, fontWeight = FontWeight.Bold,
                color = if (urgent) C.accentRed.toColor() else C.accentGreen.toColor(),
                fontFamily = LocalGameFont.current
            )
            Text(
                n.month, fontSize = 10.sp, color = C.textFaint.toColor(),
                fontFamily = LocalGameFont.current
            )
        }
        Text(
            n.headline, fontSize = 14.sp, fontWeight = FontWeight.Bold,
            color = C.textDark.toColor(), fontFamily = LocalGameFont.current,
            modifier = Modifier.padding(top = 2.dp)
        )
        Text(
            n.body, fontSize = 11.sp, color = C.textMid.toColor(),
            fontFamily = LocalGameFont.current, lineHeight = 17.sp,
            modifier = Modifier.padding(top = 2.dp)
        )
    }
}

@Suppress("unused")
private val unusedColor: Color = Color.Transparent

File: scripts/Screens/NewspaperScreen.lua
-- ============================================================================
-- NewspaperScreen — 城市简报（新闻回顾 flavor，纯 UI）
-- ============================================================================

local UI = require("urhox-libs/UI")
local Config = require("Config")
local GameData = require("GameData")
local World = require("world.World")

local NewspaperScreen = {}
local C = Config.COLORS

local function newsCard(n)
  local urgent = n.tag == "头条"
  return UI.Panel {
    width = "100%",
    flexDirection = "column",
    padding = { top = 10, bottom = 10, left = 12, right = 12 },
    backgroundColor = C.chipBg,
    borderRadius = 12,
    borderWidth = 1,
    borderColor = urgent and C.accentRed or C.border2,
    borderLeftWidth = 3,
    borderLeftColor = urgent and C.accentRed or C.accentGreen,
    marginBottom = 8,
    children = {
      UI.Panel {
        flexDirection = "row", justifyContent = "space-between", width = "100%",
        children = {
          UI.Label { text = n.tag or "快讯", fontSize = 10, fontWeight = "bold",
                     fontColor = urgent and C.accentRed or C.accentGreen },
          UI.Label { text = n.month, fontSize = 10, fontColor = C.textFaint },
        },
      },
      UI.Label { text = n.headline, fontSize = 14, fontWeight = "bold", fontColor = C.textDark, marginTop = 2 },
      UI.Label { text = n.body, fontSize = 11, fontColor = C.textMid, lineHeight = 1.5, marginTop = 2 },
    },
  }
end

function NewspaperScreen.create(Navigator)
  return UI.Panel {
    id = "screen_newspaper",
    width = "100%",
    height = "100%",
    flexDirection = "column",
    visible = false,
    backgroundColor = C.uiBackdrop,
    pointerEvents = "auto",
    children = {
      -- 顶栏
      UI.Panel {
        flexDirection = "row",
        alignItems = "center",
        justifyContent = "space-between",
        width = "100%",
        padding = { top = 12, bottom = 12, left = 14, right = 14 },
        children = {
          UI.Panel {
            id = "paperBack",
            flexDirection = "row", alignItems = "center", gap = 4,
            padding = { top = 8, bottom = 8, left = 14, right = 14 },
            backgroundColor = C.panelWhite, borderRadius = 18, pointerEvents = "auto",
            onClick = function() Navigator.go("map") end,
            boxShadow = { { x = 0, y = 2, blur = 8, color = { 90, 100, 90, 80 } } },
            children = {
              UI.Label { text = "← 返回地图", fontSize = 13, fontWeight = "bold", fontColor = C.textDark },
            },
          },
          UI.Label { text = "晨光简报", fontSize = 16, fontWeight = "bold", fontColor = C.textDark },
          UI.Panel { width = 84 },  -- 占位平衡
        },
      },

      -- 概览卡
      UI.Panel {
        id = "paperStats",
        width = "100%",
        flexDirection = "column",
        padding = { top = 12, bottom = 12, left = 16, right = 16 },
        backgroundColor = C.panelWhite, borderRadius = 16,
        margin = { top = 0, bottom = 8, left = 14, right = 14 },
      },

      -- 新闻滚动
      UI.ScrollView {
        flex = 1,
        width = "100%",
        padding = { top = 4, bottom = 16, left = 14, right = 14 },
        children = {
          UI.Panel { id = "paperNews", flexDirection = "column", width = "100%", children = {} },
        },
      },
    },
  }
end

function NewspaperScreen.onShow()
  local root = UI.GetRoot()
  if not root then return end
  local s = GameData.current
  local level = World.cityLevel()

  local stats = root:FindById("paperStats")
  if stats then
    stats:RemoveAllChildren()
    stats:AddChild(UI.Label {
      text = level.name .. " · " .. GameData.monthLabel() .. " 概况",
      fontSize = 13, fontWeight = "bold", fontColor = C.textDark, marginBottom = 6,
    })
    stats:AddChild(UI.Panel {
      flexDirection = "row", justifyContent = "space-between", width = "100%",
      children = {
        UI.Label { text = "财政 ¥" .. tostring(math.floor(s.funds)) .. "万", fontSize = 12, fontColor = C.accentGold },
        UI.Label { text = "人口 " .. tostring(s.population), fontSize = 12, fontColor = C.textDark },
        UI.Label { text = "满意度 " .. tostring(math.floor(s.happiness)), fontSize = 12,
                   fontColor = s.happiness >= 60 and C.accentGreen or C.accentRed },
      },
    })
  end

  local area = root:FindById("paperNews")
  if area then
    area:RemoveAllChildren()
    local news = GameData.recentNews(20)
    if #news == 0 then
      area:AddChild(newsCard({ tag = "头条", headline = "暂无报道", body = "城市还在建设中。", month = GameData.monthLabel() }))
    else
      for i = #news, 1, -1 do
        area:AddChild(newsCard(news[i]))
      end
    end
  end
end

return NewspaperScreen
